File: sync/rest/maps/create-map-item/create-map-item.5.x.rb
require 'twilio-ruby'

# Initialize the client
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
client = Twilio::REST::Client.new(account_sid, auth_token)

# Retreive the service
service = client.sync.v1.services('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')

# Create a Map Item
response = service.sync_maps('Players').sync_map_items.create(
  key:  'steph_curry',
  data: "{ 'name': 'Stephen Curry', 'level': '30', 'username':'spicy_curry'}",
  ttl: 864000  # expires in 10 days
)

puts response

File: sync/rest/maps/create-map-item/create-map-item.5.x.js
// To set up environmental variables, see http://twil.io/secure
const accountSid = process.env.TWILIO_ACCOUNT_SID;
const authToken = process.env.TWILIO_AUTH_TOKEN;
const Twilio = require('twilio').Twilio;

const client = new Twilio(accountSid, authToken);
const service = client.sync.services('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX');

service
  .syncMaps('Players')
  .syncMapItems.create({
    key: 'steph_curry',
    data: {
      level: 30,
      name: 'Stephen Curry',
      username: 'spicy_curry',
    },
    ttl: 864000 // expires in 10 days
  })
  .then(response => {
    console.log(response);
  })
  .catch(error => {
    console.log(error);
  });
